### PR TITLE
fix(back, front): mentions in recent activities

### PR DESCRIPTION
**Pull request by Millefeuille42**
#6272
#6279

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [ ] Manual, quality tests have been done

### CODE REVIEW

**Review by Millefeuille42 — COMMENTED** [review]
Ok for me (for everything that is not the tests)

**Review by Skylsmoi — APPROVED** [review]
approving the functional test

File: backend/tracim_backend/lib/core/user.py [hunk 1]
```diff
@@ -243,9 +246,70 @@ def get_members_of_workspaces(self, workspace_ids: typing.List[int]) -> typing.L
         )
         return [item[0] for item in user_ids_in_workspaces_tuples]
 
+    def get_common_user_workspaces_for_user(self, user_id: int) -> List[Tuple[Any, ...]]:
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/core/user.py:249]: what is a "common" user?
Is it a concept that already exists in backend?

File: backend/tracim_backend/lib/core/user.py [hunk 2]
```diff
@@ -243,9 +246,70 @@ def get_members_of_workspaces(self, workspace_ids: typing.List[int]) -> typing.L
...
+        """
+        Return list of UserRoleInWorkspace that are in common with the provided user
+        :param user_id: id of the user to get data from
+        :return: List of found UserRoleInWorkspace in common with the provided user
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/core/user.py:253]: if it returns a list of UserRoleInWorkspace, shouldn't the typing reflect that?

File: backend/tracim_backend/lib/core/user.py [hunk 3]
```diff
@@ -243,9 +246,70 @@ def get_members_of_workspaces(self, workspace_ids: typing.List[int]) -> typing.L
...
+                current_index += 1
+                user = User()
+                user.user_id = user_workspace_setting[0]
+                user.username = user_workspace_setting[1]
+                user.display_name = user_workspace_setting[2]
+                user.workspace_ids = [user_workspace_setting[3]]
+                users.append(user)
+                prev_user_id = user.user_id
+                continue
+            # If the user_id is the same as the precedent one, simply add the workspace
```
Comment (Skylsmoi) [inline backend/tracim_backend/lib/core/user.py:305]: avoid usage of words like "simply", "just", "easily" and such in documentations

File: backend/tracim_backend/views/core_api/schemas.py [hunk 4]
```diff
@@ -1058,7 +1059,7 @@ def make_path_object(self, data: typing.Dict[str, typing.Any]) -> object:
 
 
 class KnownMembersQuerySchema(marshmallow.Schema):
-    acp = StrippedString(example="test", description="search text to query", required=True)
+    acp = StrippedString(example="test", description="search text to query", required=False)
```
Comment (Skylsmoi) [inline backend/tracim_backend/views/core_api/schemas.py:1062]: We need to log this type of change somewhere to not forget it during the writing the changelog

I'll check how we use to do so

**Review by Skylsmoi — COMMENTED** [review]
review for backend part only

File: backend/tracim_backend/lib/core/user.py (hunk 1 above)
Comment (Skylsmoi) [reply backend/tracim_backend/lib/core/user.py:249]:
```suggestion
    def get_users_in_common_with_user_workspace(self, user_id: int) -> List[Tuple[Any, ...]]:
```

File: backend/tracim_backend/lib/core/user.py (hunk 2 above)
Comment (Millefeuille42) [reply backend/tracim_backend/lib/core/user.py]: Changed in last commit. Return type is the same but I changed the description

File: backend/tracim_backend/lib/core/user.py (hunk 3 above)
Comment (Millefeuille42) [reply backend/tracim_backend/lib/core/user.py]: Removed in last commit

File: backend/tracim_backend/views/core_api/schemas.py (hunk 4 above)
Comment (Skylsmoi) [reply backend/tracim_backend/views/core_api/schemas.py:1062]: documented in https://github.com/tracim/tracim/issues/6280

File: backend/tracim_backend/lib/core/user.py (hunk 1 above)
Comment (Skylsmoi) [reply backend/tracim_backend/lib/core/user.py]: fixed

File: backend/tracim_backend/lib/core/user.py (hunk 2 above)
Comment (Skylsmoi) [reply backend/tracim_backend/lib/core/user.py]: fixed

File: backend/tracim_backend/lib/core/user.py (hunk 3 above)
Comment (Skylsmoi) [reply backend/tracim_backend/lib/core/user.py]: fixed

File: frontend/src/action-creator.sync.js [hunk 5]
```diff
@@ -113,14 +116,20 @@ export const updateWorkspaceDetail = workspaceDetail => ({ type: `${UPDATE}/${WO
...
-export const WORKSPACE_MEMBER = `${WORKSPACE}/Member`
+const MEMBER = 'Member'
+export const WORKSPACE_MEMBER = `${WORKSPACE}/${MEMBER}`
 export const WORKSPACE_MEMBER_LIST = `${WORKSPACE_MEMBER}/List`
 export const setWorkspaceMemberList = workspaceMemberList => ({ type: `${SET}/${WORKSPACE_MEMBER_LIST}`, workspaceMemberList })
+// INFO - CH - 2023-11-22 - WORKSPACE_MEMBER_ADD is used by api function postWorkspaceMember()
+// but nothing in bind to it
 export const WORKSPACE_MEMBER_ADD = `${WORKSPACE_MEMBER}/${ADD}`
+// INFO - CH - 2023-11-22 - WORKSPACE_MEMBER_ADD is used by api function deleteWorkspaceMember()
+// but nothing in bind to it
```
Comment (Millefeuille42) [inline frontend/src/action-creator.sync.js:127]: > but nothing in bind to it

wouldn't it be "but nothing is bound to it" ?

File: frontend/test/setup.js [hunk 6]
```diff
@@ -47,4 +47,4 @@ if (!global.window && !global.document) {
 
 Enzyme.configure({ adapter: new Adapter() })
 chai.use(chaiEnzyme())
-sinon.stub(console, 'log')
+// sinon.stub(console, 'log')
```
Comment (Millefeuille42) [inline frontend/test/setup.js:50]: Is this left on purpose ?

Comment (Skylsmoi) [reply frontend/test/setup.js:50]: no, will fix

File: frontend/src/action-creator.sync.js (hunk 5 above)
Comment (Skylsmoi) [reply frontend/src/action-creator.sync.js]: will fix

File: frontend/test/setup.js (hunk 6 above)
Comment (Skylsmoi) [reply frontend/test/setup.js]: fixed

**Comment by Skylsmoi** [comment]
I skipped backend test and forntend test to save bandwidth because they already passed

**Pull request merged**